### ISSUE TITLE
Proposal:  make Trema dual licensed under the MIT or GPL Version2 licenses

### ISSUE THREAD

**Issue by otahi** [trema/trema#429]
## tl;dr

For Trema to be used widely and actively, the license of Trema framework is expected to be suitable for any Trema applications. However, currently, it might be considered that applications should be treated as the GPLv2 (because applications and Trema share memory space at runtime.) So I propose to change Trema's License to MIT/GPLv2 dual-license.

---

I propose the Trema codes such as the framework, samples, applications and related tools making dual-license of  GPLv2 and MIT to provide the freedom of distribution of SDN applications that developed using Trema framework.

I think it is the unspoken understanding that the applications developed using the Trema(or the other Ruby frameworks) aren't to be always necessary to accord with the license of the framework, similarly to the gcc toolchains' runtime exception. The objects built by the gcc aren't restricted by GPLv2.

Because essentially an interpreter type language like Ruby can't help using functions of framework dynamically, it's natural that the applications' OSS to be handled like a runtime exception of gcc.
The person who has developed applications should be able to choose a license freely. I believe that there are no most people in opposition to this.

However, there might be a person or a group who thinks the application should be treated as the GPLv2. As I described above, because an interpreter type language like Ruby can't help reading other objects in its memory space dynamically to run a code, such the person or the group insists that the whole program is GPLv2 if a piece of the GPLv2 object just exists in the same memory space of the application process.

Of course, it isn't impossible to develop a program as there are no objects of GPLv2 in memory space of an identical process. However, that makes to use Trema (or the other Ruby frameworks) meaningless. This is reinventing the wheel.

I believe that the member's purpose of Trema community is to promote innovation of the network by developing SDN/OpenFlow applications using provided framework. SDN application developers are frightened by a threat of the license and thereby "reinventing the wheel", that is to say, programs can't be made readily.

Yukihiro Matsumoto, the designer of Ruby language, also changed the license of Ruby to address the concerns of similar problems. It's very natural that Trema accords with Ruby's principle. 
To defend a right of the person who would like to use GPLv2, I'm convinced that it's reasonable choice to make it a dual license. It will become easy to combine with  licensed under GPLv2.

I would like to restate my proposal. I propose the Trema codes such as the framework, samples, applications and related tools making dual-license of  GPLv2 and MIT. The license clauses in detail are as follows.
- http://www.gnu.org/licenses/gpl-2.0.html 
- https://opensource.org/licenses/MIT

I believe that my suggestion will be supported by every understanding SDN application developer. Please advise on correction of license provisions and others.

**Comment by p000458368** [comment]
I fully support your suggestions . Thanks for the great suggestion .

**Comment by tj-baldwin** [comment]
I can not expect that the SDN / Trema developers and contributors who have common sense to reject this proposal .I believe that this proposal should be supported .
